docs/system_overview/configuration.md

```diff
@@ -623,11 +741,25 @@ This is the main iRODS configuration file defining the iRODS environment. Any ch
     // certificates. Use the ‘c_rehash’ utility to create the necessary links.
     "irods_ssl_ca_certificate_path": "",
 
+    // (Optional)
+    // Defines the level of server certificate authentication to perform.
+    //
+    // The following values are supported: none, cert, hostname
+    //
+    // When set to "none", authentication is skipped.
+    //
+    // When set to "cert", the server will verify that the certificate was signed by a trusted CA.
+    //
+    // When set to "hostname", the server will do everything defined by the "cert" level and then verify
+    // that the FQDN of the iRODS server matches either the common name or one of the subjectAltNames in
+    // the certificate.
+    "irods_ssl_verify_server": "hostname",
```

Comment:
Confirm `hostname` is the actual default.

Reply:
Out of scope for this PR. Will open an issue for this.

Reply:
Did some debugging with @alanking and discovered the following:
- The TLS code treats 0 as an error and 1 as success
- `hostname` is an implicit default - i.e. the value is never explicitly set to `hostname`
- `cert` isn't a real value - i.e. any value other than `none`, `hostname`, or empty string results in `cert` behavior
  - The iRODS 5 schema limits the values to `none`, `hostname`, and `cert` which is good

We think `hostname` is indeed the default. I'm going adjust the wording to use `implicit`.

Reply:
Will do.

Reply:
Thinking about this a little more, the current wording is fine because iRODS 5 doesn't allow anything other than the three values.

The behavior should be documented in the server code.